### PR TITLE
133 rotate ep

### PR DESCRIPTION
**Pull request by kalebdavis**
fixes https://github.com/PagerNation/PagerNation/issues/133

### CODE REVIEW

File: server/services/group.js [hunk 1]
```diff
@@ -89,9 +98,42 @@ function addAdmin(groupName, userId) {
     .then(validatedUserId => Group.addAdmin(groupName, validatedUserId));
 }
 
+function scheduleEPRotation(group) {
+  const rotationInterval = group.escalationPolicy.rotationIntervalInDays;
+  const nextRotateDate = buildRotateDate(group.lastRotated, rotationInterval);
+  return new Promise((resolve) => {
```
Comment (jarrydlee) [inline server/services/group.js]: If you need a Promise in the return, you can just do `return Promise.resolve(...)` and include the data you need to be wrapped in a promise.

File: server/services/group.js [hunk 2]
```diff
@@ -89,9 +98,42 @@ function addAdmin(groupName, userId) {
...
+    resolve({ nextRotateDate, group });
+  });
+}
+
+function rotateEscalationPolicy(group) {
+  const subscribers = group.escalationPolicy.subscribers;
+  const newFirstUser = subscribers.pop();
+  subscribers.unshift(newFirstUser);
+  for (let i = 0; i < subscribers.length; i++) {
+    subscribers[i] = subscribers[i].toString();
```
Comment (jarrydlee) [inline server/services/group.js]: You can use a `.map()` here

File: server/services/group.js [hunk 3]
```diff
@@ -89,9 +98,42 @@ function addAdmin(groupName, userId) {
...
+  }
+  const groupUpdates = { lastRotated: new Date() };
+  return scheduleEPRotation(group)
+    .then(() => updateGroup(group.name, groupUpdates))
+    .then(() => updateEscalationPolicy(group.name, { subscribers }));
+}
+
+// Internal Helper Function
```
Comment (jarrydlee) [inline server/services/group.js]: Do we need the doc?

File: server/tests/models/group.test.js [hunk 4]
```diff
@@ -30,9 +30,31 @@ describe('## Group Model', () => {
...
+  context('# with a passed in date', () => {
+    const date = new Date(2018, 8, 15, 18, 56, 30);
+    const groupData = {
+      name: 'Wondertwins',
+      users: [userObjectId],
+      lastRotated: date
+    };
+
+    it('should create a new group with that date', (done) => {
+      new Group(groupData).save((err, newGroup) => {
```
Comment (jarrydlee) [inline server/tests/models/group.test.js]: Could you use fixtures here?

File: server/tests/helpers/date_helper.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
```
Comment (jarrydlee) [inline server/tests/helpers/date_helper.js]: Can you camelCase the file name

File: server/services/group.js (hunk 3 above)
Comment (kalebdavis) [inline server/services/group.js]: I included it to make it clear that it wouldn't be called by any external service

File: server/tests/helpers/date_helper.js (hunk 5 above)
Comment (kalebdavis) [inline server/tests/helpers/date_helper.js]: I followed the convention of the helper that already exists, so if I change this one I'll change the other two. 

We don't have a convention for all files, sometimes they're snake and sometimes camelCased, so which do we want to go with?

**Comment by kalebdavis** [comment]
The linter runs in CI, so it's already be run and linted. Were there specific files that you noticed don't look linted?

File: server/tests/helpers/date_helper.js (hunk 5 above)
Comment (jarrydlee) [inline server/tests/helpers/date_helper.js]: Is it just spec_helper and this? Let's go with camelCase to stay consistent.

**Comment by jarrydlee** [comment]
Yeah there were a couple times the spacing looked off, I'll comment on one.

**Comment by jarrydlee** [comment]
It looks like you made a change after/during my review and the spacing issues I saw are gone.

File: index.js [hunk 6]
```diff
@@ -15,6 +16,7 @@ const debug = require('debug')('escalator-api:index');
 
 // listen on port config.port
 app.listen(config.port, () => {
+  groupLoader.bulkScheduleEPRotation();
```
Comment (kalebdavis) [inline index.js]: Just tested it, apparently not.

File: server/middlewares/group.js [hunk 7]
```diff
@@ -10,6 +10,14 @@ function isGroupAdmin(req, res, next) {
   res.status(httpStatus.UNAUTHORIZED).send();
 }
 
+function isSysAdmin(req, res, next) {
```
Comment (jarrydlee) [inline server/middlewares/group.js]: Should this be in the group middleware directory?

**Pull request merged**